File: go/concepts/input_output.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

// Input & Output
func demonstrateIO() {
	fmt.Println("\n=== Input and Output ===")

	// Create a new reader to read from standard input.
	reader := bufio.NewReader(os.Stdin)

	fmt.Print("Enter your name: ")

	// Read input until the user presses Enter.
	// Go requires explicit error handling.
	userInput, err := reader.ReadString('\n')
	if err != nil {
		fmt.Println("An error occurred while reading input:", err)
		return
	}

	// Remove whitespace and the newline character from the input.
	userInput = strings.TrimSpace(userInput)

	fmt.Printf("Hello, %s!\n", userInput)
}

File: cpp/concepts/input_output.cpp
#include <iostream>
#include <string>
#include "concepts.hpp"

// Input & Output
void demonstrateIO() {
    std::cout << "\n=== Input and Output ===" << std::endl;

    std::string user_input;
    std::cout << "Enter your name: ";

    // Use std::getline to read a full line of text, including spaces.
    std::getline(std::cin, user_input);

    std::cout << "Hello, " << user_input << "!" << std::endl;
}
